blog/_posts/GSoC 2017 project A Wrapper for the FEniCS Finite Element Toolbox.md

```diff
@@ -0,0 +1,107 @@
+
+# What is FEniCS?
+
+**[FEniCS](https://fenicsproject.org/)** describes itself as a popular open-source (LGPLv3) computing platform for solving partial differential equations (PDEs). FEniCS enables users to quickly translate scientific models into efficient finite element code. In layman's terms it lets us solve math problems.
+
+# But, how does it work? 
+
+The finite element method is a numerical method that solves partial differential equations by discretizing the domain (breaking it up into small pieces, generally triangles called finite elements) and then solving the resulting equations (which are generally a system of algebraic equations) using calculus of variations to approximate our required solution. This is nearly impossible to do by hand, so computer software is required to aid in solving it. Our wrapper provides calls to the meshing functionality, the assembly of the stiffness matrices and the solution of the variational problems. It also provides access to various *helper* functions, that make usage easier.
+
+# And, internally? 
+
+We currently provide macros to create Julia types from FEniCS classes to assist with function overloading. Apart from this we can wrap attributes and combining these two with PyCall.jl functionality we can wrap the necessary functions directly from FEniCS. We define methods for performing some linear algebra operations, and define operator functions ` +,-,* ` for various geometric objects and  UFL forms. Our exported API remains approximately the same as the pythonic FEniCS with only very slight changes.
+
+# Demonstration
+
+Below is a small demonstration of how a user would use our code to solve the Poisson equation with Dirichlet conditions. This directly mirrors one of the **[tutorials](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft01_poisson.py)** FEniCS provides 
+
+  $-\bigtriangleup(u) = f$    in the unit square
+  
+            $u = u_D$  on the boundary
+            
+  $u_D$ $=$ $1 + x^2 + 2y^2$
+  
+   $f = -6$
+
+```julia
+using FEniCS
+mesh = UnitSquareMesh(8,8) 
+V = FunctionSpace(mesh,"P",1)
+u_D = Expression("1+x[0]*x[0]+2*x[1]*x[1]", degree=2)
+u = TrialFunction(V)
+bc1 = DirichletBC(V,u_D, "on_boundary")
+v = TestFunction(V)
+f = Constant(-6.0)
+a = dot(grad(u),grad(v))*dx
+L = f*v*dx
+U = FEniCS.Function(V)
+lvsolve(a,L,U,bc1) #linear variational solver
+errornorm(u_D, U, norm="L2")
+get_array(L) #this returns an array for the stiffness matrix
+get_array(U) #this returns an array for the solution values
+vtkfile = File('poisson/solution.pvd')
+vtkfile << U.pyobject #exports the solution to a vtkfile
+
+```
+Apart from just defining the problem, we can also access and save the arrays corresponding to various variational forms. We can do this as follows :
+
+```julia
+a = dot(grad(u),grad(v))*dx #this sets up the variatonal form from the previous problem
+variational_matrix = get_array(a)
+
```

Comment:
Maybe specify here what the compatibility of the returned type is with Julia, e.g. is it an `AbstractArray`?

Reply:
@ysimillides address this and I'll approve

Reply:
has been added.